### PR TITLE
Bug 1884487 – simplify NavbarIntegration by removing viewLifecycleOwner

### PR DESCRIPTION
**Pull request by mavduevskiy**
a follow up to [this](https://github.com/mozilla-mobile/firefox-android/pull/5846)

The change makes `NavbarIntegration` implementation in line with how generally `LifecycleAwareFeature` handle flows: the pattern is save scope and manage it on lifecycle callbacks. Some examples of it (we have many more):
[Thumbnail](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/browser/thumbnails/src/main/java/mozilla/components/browser/thumbnails/BrowserThumbnails.kt#38), [CustomTabWindowFeature](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/CustomTabWindowFeature.kt#37), [TemporaryDownloadFeature](https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/temporary/TemporaryDownloadFeature.kt#66).

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1884487

### CODE REVIEW

**Comment by rahulsainani** [comment]
We should rephrase this to "since nav integration is provided the lifecycle owner", so start/stop will be called with lifecycle callbacks. I don't think this is related to scope but making sure that the flow is cancelled with the lifecycle of the caller.

**Review by rocketsroger — APPROVED** [review]

**Comment by mavduevskiy** [comment]
talked with @rahulsainani , added a description and fixed the commit message
